_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -945,6 +945,50 @@ mfxStatus VideoDECODEVP9_HW::PrepareInternalSurface(UMC::FrameMemID &mid, mfxFra
     return MFX_ERR_NONE;
 }
 
+
+static mfxStatus CheckFrameInfo(mfxFrameInfo &currInfo, mfxFrameInfo &info)
+{
+    mfxStatus sts = CheckFrameInfoCommon(&info, MFX_CODEC_VP9);
+    MFX_CHECK(sts == MFX_ERR_NONE, sts);
+
+    switch (info.FourCC)
+    {
+        case MFX_FOURCC_NV12:
+        case MFX_FOURCC_AYUV:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y410:
+#endif
+            break;
+        case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y210:
+#endif
+           if (info.Shift != 1)
+              return MFX_ERR_INCOMPATIBLE_VIDEO_PARAM;
+           break;
+        default:
+            return MFX_ERR_INCOMPATIBLE_VIDEO_PARAM;
```

Comment:
```
MFX_CHECK_STS(MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
```

Reply:
same

Reply:
what about this one?

Reply:
Ok, i'll do it. But I still think, that return ERR is more understandable, than MFX_CHECK_STS(err)!

Reply:
I agree with you, that this code looks a bit artificial. But, it allows you to trace the error to origin. Reason is that you will see which exact check was triggering and sent an error to your application. See how this mechanism works in https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/include/mfx_utils.h#L38  you need to enable ```MFX_DEBUG_TRACE``` to see the full log

Reply:
Maybe, we need to have MFX_RETURN macro for such cases. It will be much more readable.